### ISSUE TITLE
Memory usage is concentrated in gpu 0

### ISSUE THREAD

**Issue by cl886699** [ultralytics/ultralytics#1359]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I run train.py in vscode, I can only use batch=1 when training yolov8l,it seems did not using ddp mode
![image](https://user-images.githubusercontent.com/31495593/224244094-ccbd1078-0414-4127-90df-77fc50837a50.png)
![image](https://user-images.githubusercontent.com/31495593/224244236-afd3666d-fcc2-49d6-ac61-cfd42b876906.png)

/home/anaconda3/envs/th19/bin/python /home/ultralytics-main/ultralytics/yolo/v8/detect/train.py
Ultralytics YOLOv8.0.51 🚀 Python-3.8.0 torch-1.9.1+cu111 CUDA:0 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
                                                          CUDA:1 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
                                                          CUDA:2 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
                                                          CUDA:3 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
                                                          CUDA:4 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
                                                          CUDA:5 (NVIDIA GeForce RTX 2080 Ti, 11019MiB)
yolo/engine/trainer: task=detect, mode=train, model=yolov8l.yaml, data=/home/ultralytics-main/ultralytics/yolo/v8/tstcd.yaml, epochs=100, patience=50, batch=6, imgsz=640, save=True, save_period=-1, cache=False, device=0,1,2,3,4,5, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, min_memory=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=True, tracker=botsort.yaml, save_dir=runs/detect/train
Running DDP command ['/home/anaconda3/envs/th19/bin/python', '-m', 'torch.distributed.run', '--nproc_per_node', '6', '--master_port', '35702', '/home/ultralytics-main/ultralytics/yolo/v8/detect/train.py', 'task=detect', 'mode=train', 'model=yolov8l.yaml', 'data=/home/ultralytics-main/ultralytics/yolo/v8/tstcd.yaml', 'epochs=100', 'patience=50', 'batch=6', 'imgsz=640', 'save=True', 'save_period=-1', 'cache=False', 'device=0,1,2,3,4,5', 'workers=8', 'project=None', 'name=None', 'exist_ok=False', 'pretrained=False', 'optimizer=SGD', 'verbose=True', 'seed=0', 'deterministic=True', 'single_cls=False', 'image_weights=False', 'rect=False', 'cos_lr=False', 'close_mosaic=10', 'resume=False', 'min_memory=False', 'overlap_mask=True', 'mask_ratio=4', 'dropout=0.0', 'val=True', 'split=val', 'save_json=False', 'save_hybrid=False', 'conf=None', 'iou=0.7', 'max_det=300', 'half=False', 'dnn=False', 'plots=True', 'source=None', 'show=False', 'save_txt=False', 'save_conf=False', 'save_crop=False', 'hide_labels=False', 'hide_conf=False', 'vid_stride=1', 'line_thickness=3', 'visualize=False', 'augment=False', 'agnostic_nms=False', 'classes=None', 'retina_masks=False', 'boxes=True', 'format=torchscript', 'keras=False', 'optimize=False', 'int8=False', 'dynamic=False', 'simplify=False', 'opset=None', 'workspace=4', 'nms=False', 'lr0=0.01', 'lrf=0.01', 'momentum=0.937', 'weight_decay=0.0005', 'warmup_epochs=3.0', 'warmup_momentum=0.8', 'warmup_bias_lr=0.1', 'box=7.5', 'cls=0.5', 'dfl=1.5', 'fl_gamma=0.0', 'label_smoothing=0.0', 'nbs=64', 'hsv_h=0.015', 'hsv_s=0.7', 'hsv_v=0.4', 'degrees=0.0', 'translate=0.1', 'scale=0.5', 'shear=0.0', 'perspective=0.0', 'flipud=0.0', 'fliplr=0.5', 'mosaic=1.0', 'mixup=0.0', 'copy_paste=0.0', 'cfg=None', 'v5loader=True', 'tracker=botsort.yaml']
WARNING:__main__:*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
DDP settings: RANK 0, WORLD_SIZE 6, DEVICE cuda:0
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks failed ❌. Anomalies were detected with AMP on your system that may lead to NaN losses or zero-mAP results, so AMP will be disabled during training.
optimizer: SGD(lr=0.01) with parameter groups 97 weight(decay=0.0), 104 weight(decay=0.000515625), 103 bias

**Comment by nuttass** [comment]
I have the same problem when attempting to train YOLOv8x6 using 8 GPUs. All memory usage is concentrated in GPU 0, which means that it is only possible to train using batch size 4. I upgraded to ultralytics 8.0.52, but it did not solve the problem.

**Comment by pderrenger** [comment]
@nuttass Thank you for the update. It seems that the issue persists even with the latest version. We will investigate further to identify and address the root cause. In the meantime, I appreciate your patience and understanding as we work to resolve this issue.
